src/config/yaml_baking_conf_parser.py

```diff
@@ -207,7 +207,7 @@ def validate_payment_address(self, conf_obj):
             )
 
         if len(pymnt_addr) == PKH_LENGHT and pymnt_addr.startswith("tz"):
-            if not self.dry_run:
+            if not self.dry_run == DRY_RUN['NO_SIGNER'] or DRY_RUN['NO_SIGNER_NO_CONSUMER']:
```

Comment:
should swap this to just be if dry run == signer.

Reply:
that is expected behaviour. 
In PR (615)[https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/pull/615] It was assumed that we should never use signers in. 

I'm now saying if dry_run is not NO_SIGNER or NO_SIGNER_NO_CONSUMER please run check_pkh_known_by_signer

Reply:
I am sorry, in case you do -D and no argument it goes to default? What happens with the field if you don`t use -D at all? I think the confusing part is the implcit behavior of it is either seen and evaluated as a string which defines the type of dry_run or is not existent and therefore a no dry_run at all. I think the field should explicitly express the behavior?